docs/events.md

```diff
@@ -0,0 +1,147 @@
+# PHP Events
+
+## Conversation related events
+
+### Created conversation
+
+* Event class: `OCA\Talk\Events\RoomEvent`
+* Event name: `OCA\Talk\Room::createdRoom`
+
+### Create token for conversation
+
+* Event class: `OCA\Talk\Events\CreateRoomTokenEvent`
+* Event name: `OCA\Talk\Manager::generateNewToken`
+
+### Set name
+
+* Event class: `OCA\Talk\Events\ModifyRoomEvent`
+* Pre-event name: `OCA\Talk\Room::preSetName`
+* Post-event name: `OCA\Talk\Room::postSetName`
+
+### Set password
+
+* Event class: `OCA\Talk\Events\ModifyRoomEvent`
+  - No old value is provided
+* Pre-event name: `OCA\Talk\Room::preSetPassword`
+* Post-event name: `OCA\Talk\Room::postSetPassword`
+
+### Set type
+
+* Event class: `OCA\Talk\Events\ModifyRoomEvent`
+* Pre-event name: `OCA\Talk\Room::preSetType`
+* Post-event name: `OCA\Talk\Room::postSetType`
+
+### Set read-only
+
+* Event class: `OCA\Talk\Events\ModifyRoomEvent`
+* Pre-event name: `OCA\Talk\Room::preSetReadOnly`
+* Post-event name: `OCA\Talk\Room::postSetReadOnly`
+
+### Set lobby
+
+* Event class: `OCA\Talk\Events\ModifyLobbyEvent`
+* Pre-event name: `OCA\Talk\Room::preSetLobbyState`
+* Post-event name: `OCA\Talk\Room::postSetLobbyState`
+
+### Clean up guests
+
+* Event class: `OCA\Talk\Events\RoomEvent`
+* Pre-event name: `OCA\Talk\Room::preCleanGuests`
+* Post-event name: `OCA\Talk\Room::postCleanGuests`
+
+### Verify password
+
+* Event class: `OCA\Talk\Events\VerifyRoomPasswordEvent`
+* Event name: `OCA\Talk\Room::verifyPassword`
+
+### Delete conversation
+
+* Event class: `OCA\Talk\Events\RoomEvent`
+* Pre-event name: `OCA\Talk\Room::preDeleteRoom`
+* Post-event name: `OCA\Talk\Room::postDeleteRoom`
+
+## Participant related events
+
+### Add participants
+
+* Event class: `OCA\Talk\Events\AddParticipantsEvent`
+* Pre-event name: `OCA\Talk\Room::preAddUsers`
+* Post-event name: `OCA\Talk\Room::postAddUsers`
```

Comment:
For consistency with legacy hooks and events I'm fine with *pre* and *post* but we could also name these something like *BeforeUserAdded* and *AfterUserAdded*.

Reply:
Makes sense if that is the new guide line.
Since we change the event class anyway it will break with all stuff.
But at least you knew it broke, but well so far the events were not publicly announced anyway

Reply:
It was more of a spontaneous idea but after a few days I still like it. I'll try to find some time to write down some guidelines for the events in our documentation.